### PR TITLE
Use BipedalLocomotion::GlobalCoPEvaluator to compute the global CoP

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
With this PR `BipedalLocomotion::GlobalCoPEvaluator` is used to compute the global CoP for the walking controller. Thanks to this we centralize the code used to compute the quantity in all the projects

### CODE REVIEW

**Comment by GiulioRomualdi** [comment]
@g-cervettini, here you can find  how to use  `BipedalLocomotion::GlobalCoPEvaluator`

File: src/WalkingModule/src/Module.cpp [hunk 1]
```diff
@@ -226,6 +196,16 @@ bool WalkingModule::configure(yarp::os::ResourceFinder &rf)
         return false;
     }
 
+    yInfo() << rf.toString();
```
[the remark this answers is not in the document]
Comment (GiulioRomualdi) [reply src/WalkingModule/src/Module.cpp:199]: Leftover :) I'm removing it

File: src/WalkingModule/src/Module.cpp [hunk 2]
```diff
@@ -1517,11 +1402,26 @@ bool WalkingModule::startWalking()
...
+        return false;
+    }
+
+    if (!m_globalCoPEvaluator.advance())
     {
-        yError() << "[WalkingModule::startWalking] Unable to evaluate the ZMP.";
+        yError() << "[WalkingModule::startWalking] Unable to compute the global ZMP.";
         return false;
     }
+    iDynTree::toEigen(measuredZMP) = m_globalCoPEvaluator.getOutput().head<2>();
```
[the remark this answers is not in the document]
Comment (GiulioRomualdi) [reply src/WalkingModule/src/Module.cpp]: Done in 499eb1c

File: src/WalkingModule/src/Module.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (GiulioRomualdi) [reply src/WalkingModule/src/Module.cpp]: Done in 499eb1c

File: src/WalkingModule/src/Module.cpp [hunk 3]
```diff
@@ -693,12 +695,22 @@ bool WalkingModule::updateModule()
...
         }
 
+        // remove the ZMP offset if required
+        if (m_removeZMPOffset)
+        {
+            // remove rotated offset
+            iDynTree::Rotation yawRotation = this->computeAverageYawRotationFromPlannedFeet();
+            m_zmpOffset = yawRotation * m_zmpOffsetLocal;
+            iDynTree::toEigen(measuredZMP) += iDynTree::toEigen(m_zmpOffset).head<2>();
+        }
```
Comment (S-Dafarra) [inline src/WalkingModule/src/Module.cpp:712]: Isn't also this supposed to go in the function?

Comment (GiulioRomualdi) [reply src/WalkingModule/src/Module.cpp:712]: This is actually this required only in the update module since in start walking we need to compute it

Comment (S-Dafarra) [reply src/WalkingModule/src/Module.cpp:712]: Ah I see, sounds good then!

**Review by S-Dafarra — APPROVED** [review]

**Pull request merged**